iceoryx_hoofs/source/posix_wrapper/posix_access_rights.cpp

```diff
@@ -58,7 +58,7 @@ bool PosixGroup::operator==(const PosixGroup& other) const noexcept
 
 PosixGroup PosixGroup::getGroupOfCurrentProcess() noexcept
 {
-    return PosixGroup(getegid());
+    return PosixGroup(getgid());
```

Comment:
To be honest, I don't yet understand the difference between real and effective GID. If it's not too complex, can you please explain why this change is necessary? Otherwise I'll continue with my research.

Reply:
According to https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap03.html#tag_03_140:
- real GID identifies the group of the user at the time of process creation
- it can be changed by a process to perform tasks as a member of another group, leading to the effective GID

Reply:
@FerdinandSpitzschnueffler awesome thank you.